File: backend/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Product extends Model
{
    use HasFactory;

    // Set nama tabel secara eksplisit jika nama model tidak sesuai konvensi plural Laravel (Product -> products)
    protected $table = 'products';

    // Set primary key secara eksplisit jika bukan 'id'
    protected $primaryKey = 'product_id';

    // Primary key bukan auto-incrementing atau bukan integer, set to false
    // (sesuaikan jika product_id Anda memang SERIAL atau auto-increment)
    public $incrementing = true; // Jika product_id SERIAL
    protected $keyType = 'int'; // Jika product_id INTEGER

    // Kolom yang dapat diisi secara massal
    protected $fillable = [
        'product_id', // Jika Anda mengelola ID secara manual saat POST/PUT
        'gender_id',
        'product_type_id',
        'colour_id',
        'usage_id',
        'title',
        'image_url',
    ];

    // Definisi relasi (opsional, tapi baik untuk API)
    public function gender()
    {
        return $this->belongsTo(Gender::class);
    }

    public function productType()
    {
        return $this->belongsTo(ProductType::class);
    }

    public function colour()
    {
        return $this->belongsTo(Colour::class);
    }

    public function usage()
    {
        return $this->belongsTo(Usage::class);
    }
}

File: frontend/lib/models/product.dart
class Product {
  final int productId;
  final int genderId;
  final int productTypeId;
  final int colourId;
  final int usageId;
  final String title;
  final String imageUrl;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Product({
    required this.productId,
    required this.genderId,
    required this.productTypeId,
    required this.colourId,
    required this.usageId,
    required this.title,
    required this.imageUrl,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      productId: json['product_id'] as int,
      genderId: json['gender_id'] as int,
      productTypeId: json['product_type_id'] as int,
      colourId: json['colour_id'] as int,
      usageId: json['usage_id'] as int,
      title: json['title'] as String,
      imageUrl: json['image_url'] as String,
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'] as String)
          : null,
      updatedAt: json['updated_at'] != null
          ? DateTime.parse(json['updated_at'] as String)
          : null,
    );
  }

  // Method untuk mengubah objek Product menjadi JSON (opsional, berguna untuk POST/PUT)
  Map<String, dynamic> toJson() {
    return {
      'product_id': productId,
      'gender_id': genderId,
      'product_type_id': productTypeId,
      'colour_id': colourId,
      'usage_id': usageId,
      'title': title,
      'image_url': imageUrl,
      'created_at': createdAt?.toIso8601String(),
      'updated_at': updatedAt?.toIso8601String(),
    };
  }
}
